tests/TestHarness/Node.py

```diff
@@ -489,6 +490,32 @@ def waitForTransInBlock(self, transId, timeout=None):
         ret=Utils.waitForBool(lam, timeout)
         return ret
 
+    def checkBlocksForTransactions(self, transIds, lastBlockProcessed, endBlock):
+        for blockNum in range(endBlock, lastBlockProcessed + 1, -1):
+            block = self.processCurlCmd("trace_api", "get_block", f'{{"block_num":{blockNum}}}', silentErrors=False, exitOnError=True)
+            if block['transactions']:
+                for trx in block['transactions']:
+                    if trx['id'] in transIds:
+                        transIds.pop(trx['id'])
+        return transIds
+
+    def waitOnBlockTransactions(self, transIds, startBlock=1, timeout=30):
+        lastBlockProcessed = startBlock
+        finalBlock = sys.maxsize
+        if timeout is not None:
+            finalBlock = self.getHeadBlockNum() + timeout * 2
+        while len(transIds) > 0:
+            self.waitForHeadToAdvance()
```

Comment:
Probably don't want to add this additional wait into the algorithm for the first time through the while loop.